### PR TITLE
Example Hermes Boots

### PR DESCRIPTION
**Pull request by naakaamura**
###  What are the changes to ExampleMod?
Implementation of a "Example Hermes Boots". This example demonstrates how to add an accessory that functions the same way the Hermes Boots does.

### Do these changes need any additional documentation?
The comments I included in the code should be enough, so I believe that these changes do not require any additional documentation.

### CODE REVIEW

**Comment by JavidPack** [comment]
Does the Example Hermes Boots spawn the run dust? I remember people having issues with that. If not, make a note, I think a new hook would be needed.

**Comment by naakaamura** [comment]
> Does the Example Hermes Boots spawn the run dust? I remember people having issues with that. If not, make a note, I think a new hook would be needed.

Yes, it does.

File: ExampleMod/Items/Accessories/ExampleCloudinaBottle.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+			item.width = 16;
+			item.height = 24;
+			item.accessory = true; // Makes this item an accessory.
+			item.color = Color.Red; // Makes the item's sprite color red.
+			item.rare = ItemRarityID.Blue;
+			item.value = Item.sellPrice(silver: 50); // Sets the item sell price to 50 silver coins.
+		}
+
+		public override void UpdateAccessory(Player player, bool hideVisual) {
+			player.doubleJumpCloud = true;
```
Comment (RighteousRyan1) [inline ExampleMod/Items/Accessories/ExampleCloudinaBottle.cs:27]: In total, this just makes wearing both the could in a bottle and this useless. I would expect some sort of IL Editing to work around the jump order.

File: ExampleMod/Items/Accessories/ExampleCloudinaBottle.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
+﻿using Microsoft.Xna.Framework;
+using Terraria;
+using Terraria.ID;
+using Terraria.ModLoader;
+
+namespace ExampleMod.Items.Accessories
+{
+	public class ExampleCloudinaBottle : ModItem
+	{
+		public override string Texture => "Terraria/Item_" + ItemID.CloudinaBottle;
```
Comment (RighteousRyan1) [inline ExampleMod/Items/Accessories/ExampleCloudinaBottle.cs:10]: At least make your own texture, this just makes the two items functionally and visually identical.

File: ExampleMod/Items/Accessories/ExampleHermesBoots.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
+﻿using Microsoft.Xna.Framework;
+using Terraria;
+using Terraria.ID;
+using Terraria.ModLoader;
+
+namespace ExampleMod.Items.Accessories
+{
+	public class ExampleHermesBoots : ModItem
+	{
+		public override string Texture => "Terraria/Item_" + ItemID.HermesBoots;
```
Comment (RighteousRyan1) [inline ExampleMod/Items/Accessories/ExampleHermesBoots.cs:10]: Same here, at least try your own texture or SOMETHING different.

File: ExampleMod/Items/Accessories/ExampleCloudinaBottle.cs (hunk 2 above)
Comment (Steviegt6) [reply ExampleMod/Items/Accessories/ExampleCloudinaBottle.cs:10]: Irrelevant.

File: ExampleMod/Items/Accessories/ExampleHermesBoots.cs (hunk 3 above)
Comment (Steviegt6) [reply ExampleMod/Items/Accessories/ExampleHermesBoots.cs:10]: Irrelevant.

File: ExampleMod/Items/Accessories/ExampleCloudinaBottle.cs (hunk 1 above)
Comment (Steviegt6) [reply ExampleMod/Items/Accessories/ExampleCloudinaBottle.cs:27]: Custom double-jumps are coming in 1.4.

Comment (Steviegt6) [reply ExampleMod/Items/Accessories/ExampleCloudinaBottle.cs:27]: However, I do think a comment stating that this uses the same boolean as Cloud in a Bottle would be smart.

File: ExampleMod/Items/Accessories/ExampleHermesBoots.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+		public override void SetStaticDefaults() {
+			DisplayName.SetDefault("Example Hermes Boots");
+			Tooltip.SetDefault("The wearer can run super fast");
+		}
+
+		public override void SetDefaults() {
+			item.width = 28; 
+			item.height = 24;
+			item.accessory = true; // Makes this item an accessory.
+			item.color = Color.Orange; // Makes the item's sprite color orange.
```
Comment (Steviegt6) [inline ExampleMod/Items/Accessories/ExampleHermesBoots.cs:21]: Saying that it tints it orange would be better.

File: ExampleMod/Items/Accessories/ExampleCloudinaBottle.cs [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+		public override void SetStaticDefaults() {
+			DisplayName.SetDefault("Example Cloud in a Bottle");
+			Tooltip.SetDefault("Allows the holder to double jump");
+		}
+
+		public override void SetDefaults() {
+			item.width = 16;
+			item.height = 24;
+			item.accessory = true; // Makes this item an accessory.
+			item.color = Color.Red; // Makes the item's sprite color red.
```
Comment (Steviegt6) [inline ExampleMod/Items/Accessories/ExampleCloudinaBottle.cs:21]: Saying that it gets tinted red would be better.

**Review by Steviegt6 — COMMENTED** [review]
A couple minor notes, nothing important.

**Review by Mirsario — APPROVED** [review]
Ignoring Ryan's weirdly emotional comment, please do actually add a sprite for the item, and remove that Texture override. Then ping me or someone else to merge this. The problem is nothing more than the lambda and the override being a bit confusing for newcomers. The sprite can be literally anything (my recommendation is just a black & white version of the vanilla hermes boots.)

**Pull request merged**